### PR TITLE
Test individual product loading

### PR DESCRIPTION
**Pull request by p-datadog**
Replacement for https://github.com/DataDog/dd-trace-rb/pull/3646.

This PR tests that each product can be loaded by itself, and repairs appsec to be loadable without tracing having already been loaded.

The tests are redone to use system rather than forking and no longer require a separate job.

### CODE REVIEW

**Review by marcotc — APPROVED** [review]
Awesome clean up!

File: spec/loading_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+require 'English'
```
Comment (TonyCTHsu) [inline spec/loading_spec.rb:1]: Do we need this `require 'English'` ?

**Review by TonyCTHsu — COMMENTED** [review]
How is this PR different from https://github.com/DataDog/dd-trace-rb/pull/3646 ?

**Comment by p-datadog** [comment]
@TonyCTHsu 

The tests are redone to use `system` rather than forking and no longer require a separate job.

**Comment by p-datadog** [comment]
Also the PR is a new one because the branch is in the main repo, not in a fork for CI reasons.

File: spec/loading_spec.rb (hunk 1 above)
Comment (p-datadog) [reply spec/loading_spec.rb]: I don't think it is necessary now, removed.

**Review by TonyCTHsu — APPROVED** [review]
The test LGTM. Not entirely sure about the appsec changes.

File: lib/datadog/appsec/contrib/sinatra/patcher.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-require_relative '../../../tracing/contrib/rack/middlewares'
+require_relative '../../../tracing/contrib'
```
Comment (p-datadog) [inline lib/datadog/appsec/contrib/sinatra/patcher.rb:3]: Individual contribs require the REGISTRY constant defined by top-level contrib.

File: lib/datadog/appsec/extensions.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 require_relative 'configuration'
+require_relative '../core/configuration'
```
Comment (p-datadog) [inline lib/datadog/appsec/extensions.rb:4]: There is a dependency on Core::Configuration in this file.

**Comment by p-datadog** [comment]
@TonyCTHsu I added comments for the appsec changes, if there is a different solution that would also be fine with me.

File: spec/loading_spec.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+          unless defined?(#{const})
+            exit 1
+          end
+
+          exit 0
+        E
+      end
+
+      it 'loads successfully by itself' do
+        rv = system("ruby -e #{Shellwords.shellescape(code)}")
```
Comment (ivoanjo) [inline spec/loading_spec.rb:33]: Hey I didn't know `shellescape` existed! This is cool!

**Pull request merged**